torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -484,10 +484,7 @@ RegisterOperators reg(
            else if (s->string() == "-inf")
              push(stack, -std::numeric_limits<double>::infinity());
            else
-             AT_ERROR(
-                 "Only 'inf' or '-inf' can be cast to a float, but got '",
-                 s->string(),
-                 "'");
+             push(stack, std::stod(s->string()));
```

Comment:
`stod` will accept things like "2.3 hello" as `2.3`, so you should also check the [`idx` parameter](http://www.cplusplus.com/reference/string/stod/) to be sure the whole string was converted.